scripts/globals/combat/physical_utilities.lua

```diff
@@ -82,85 +82,107 @@ local shieldSizeToBlockRateTable =
-            damage = math.floor(damageRatio * (kickDamage + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+            baseDamage = naturalH2hDamage + actor:getMod(xi.mod.KICK_DMG) + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(actor, target)
         else
-            damage = math.floor(damageRatio * (attacker:getWeaponDmg() + naturalH2hDamage + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+            baseDamage = naturalH2hDamage + actor:getWeaponDmg() + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(actor, target)
         end
     elseif slot == xi.slot.MAIN then
-        damage = math.floor(damageRatio * (attacker:getWeaponDmg() + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+        baseDamage = actor:getWeaponDmg() + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(actor, target)
     elseif slot == xi.slot.SUB then
-        damage = math.floor(damageRatio * (attacker:getOffhandDmg() + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+        baseDamage = actor:getOffhandDmg() + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(actor, target)
     elseif slot == xi.slot.AMMO then
-        damage = math.floor(damageRatio * (attacker:getRangedDmg() + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+        baseDamage = actor:getRangedDmg() + xi.combat.physical.calculateRangedStatFactor(actor, target)
     end
 
-    -- Scarlet Delirium
-    damage = math.floor(damage * xi.combat.damage.scarletDeliriumMultiplier(attacker))
+    damage = math.floor(baseDamage * damageRatio)
+
+    -- Scarlet Delirium multiplier.
+    damage = math.floor(damage * xi.combat.damage.scarletDeliriumMultiplier(actor))
 
-    -- Double Attack and Triple Attack
-    if physicalAttackType == xi.physicalAttackType.DOUBLE and attacker:isPC() then
-        damage = math.floor(damage * ((100 + attacker:getMod(xi.mod.DOUBLE_ATTACK_DMG)) / 100))
-    elseif physicalAttackType == xi.physicalAttackType.TRIPLE and attacker:isPC() then
-        damage = math.floor(damage * ((100 + attacker:getMod(xi.mod.TRIPLE_ATTACK_DMG)) / 100))
+    -- Double/Triple Attack multipliers.
+    local multiAttackMultiplier = 1
+    if physicalAttackType == xi.physicalAttackType.DOUBLE then
+        multiAttackMultiplier = 1 + actor:getMod(xi.mod.DOUBLE_ATTACK_DMG) / 100
+    elseif physicalAttackType == xi.physicalAttackType.TRIPLE then
+        multiAttackMultiplier = 1 + actor:getMod(xi.mod.TRIPLE_ATTACK_DMG) / 100
     end
 
-    -- Soul Eater
-    if attacker:isPC() then
-        damage = xi.combat.effect.handleSoulEater(attacker, damage)
+    damage = math.floor(damage * multiAttackMultiplier)
+
+    -- Soul Eater additive damage.
+    if actor:hasStatusEffect(xi.effect.SOULEATER) then
```

Comment:
I think we might have a function for this elsewhere (weaponskills?) so we're probably duplicating code

Reply:
yeah we do

https://github.com/LandSandBoat/server/blob/ae79697ef7cdf9fc4ad87100c36089926581dc03/scripts/globals/weaponskills.lua#L25

That version seems essentially same but it uses delHp instead of addHP as the only real notable difference

Pick one and move it out to a function so it can be shared